docs/tutorials/kafka-integration.md

```diff
@@ -0,0 +1,58 @@
+---
+sidebar_position: 18
+title: Integrating with Kafka
+description: Learn how to integrate DBOS and Kafka.
+---
+
+In this guide, you'll learn how to use DBOS transactions and workflows to process Kafka messages with exactly-once semantics.
+
+First, define your transaction or workflow. It must take in the Kafka topic, partition, and message as inputs:
+
+```javascript
+class KafkaExample{
+  @Workflow()
+  static async kafkaWorkflow(ctxt: WorkflowContext, topic: string, partition: number, message: KafkaMessage) {
+    ctxt.logger.info(`Message received: ${message.value?.toString()}`)
+  }
+}
+```
+
+Then, annotate your method with a [`@KafkaConsume`](../api-reference/decorators.md#kafka-consume) decorator specifiying which topic to consume from.
+Additionally, annotate your class with a [`@Kafka`](../api-reference/decorators.md#kafka) decorator defining which brokers to connect to.
+DBOS invokes your method exactly-once for each message sent to the topic.
+
+```javascript
+import { KafkaConfig, KafkaMessage} from "kafkajs";
+
+const kafkaConfig: KafkaConfig = {
+    brokers: ['localhost:9092']
+}
+
+@Kafka(kafkaConfig)
+class KafkaExample{
+
+  @KafkaConsume("example-topic")
+  @Workflow()
+  static async kafkaWorkflow(ctxt: WorkflowContext, topic: string, partition: number, message: KafkaMessage) {
+    ctxt.logger.info(`Message received: ${message.value?.toString()}`)
+  }
+}
+```
+
+If you need more control, you can pass detailed configurations to both the `@Kafka` and `@KafkaConsume` decorators.
```

Comment:
We should add these (`Kafka` and `KafkaConsume`) to the import?

Reply:
Might as well document the full import needed:
`import { Workflow, WorkflowContext, Kafka, KafkaConsume } from '@dbos-inc/dbos-sdk';`